libs/dal/src/repositories/message/message.schema.ts

```diff
@@ -112,10 +112,13 @@ const messageSchema = new Schema<MessageDBModel>(
       },
       data: Schema.Types.Mixed,
     },
+    expireAt: Schema.Types.Date,
   },
-  { ...schemaOptions }
+  schemaOptions
 );
 
+messageSchema.index({ expireAt: 1 }, { expires: '1h' });
```

Comment:
1 hour after its expiration date - to allow buffer between archiving and its actual deletion. When we get to the archive will rethink that time amount.

Reply:
@ainouzgali  I dont think this implementation is correct. By what the docs say this will ignore our actually ttl we set in the app and set every document to 1hour after creation. 
https://www.mongodb.com/docs/manual/tutorial/expire-data/#expire-documents-at-a-specific-clock-time

Reply:
It deletes by the expireAt field, not the 1h. But it does ignore the expire. So it does not work right. Will try a different option

Reply:
The key thing for us is that the ttl should be set by the app not the db.

Reply:
Never mind @Cliftonz , It does work as expected - deletes by the expire date we set and then waits the amount waited in the expires field.